### PR TITLE
Update interop tests to use Ice 3.8.0-alpha0

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR updates all interop tests to use Ice 3.8 (3.8.0-alpha0 for now) instead of Ice 3.7.10.

As shown by this PR, upgrading a C# app from Ice 3.7 to Ice 3.8 is doable but requires some work.

Not done by this PR:
 - fix SslTransportTests.cs (currently largely commented out)
 - setup CI

### CODE REVIEW

File: build/Versions.props [hunk 1]
```diff
@@ -4,6 +4,6 @@
         <!-- TODO update to 1.2 stable release https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3187#issuecomment-1043221630 -->
         <StyleCopAnalyzersVersion>1.2.0-beta.507</StyleCopAnalyzersVersion>
         <IceRpcVersion Condition="'$(IceRpcVersion)' == ''">0.4.0-preview1</IceRpcVersion>
-        <IceVersion Condition="'$(IceVersion)' == ''">3.7.10</IceVersion>
+        <IceVersion Condition="'$(IceVersion)' == ''">3.8.0-alpha0</IceVersion>
```
Comment (bernardnormier) [inline build/Versions.props:7]: I build/test on macos with:
```
dotnet test /p:IceHome=$HOME/builds/ice /p:IceToolsPath=$HOME/builds/ice/cpp/bin
```

You also need to "publish" locally the zeroc.ice.net NuGet package.

File: tests/Interop.Tests/ConnectionTests.cs [hunk 2]
```diff
@@ -56,6 +56,140 @@ public async Task Establish_connection_from_IceRpc_to_Ice()
         Assert.That(async () => await clientConnection.ConnectAsync(), Throws.Nothing);
     }
 
+    /// <summary>Verifies an IceRPC->Ice connection remains alive after idling for > idle timeout. That's because the
```
Comment (bernardnormier) [inline tests/Interop.Tests/ConnectionTests.cs:59]: Moved from AcmTests and reworked. Ice 3.8 and IceRPC 0.4 have essentially the same connection properties.

File: tests/Interop.Tests/ConnectionTests.cs [hunk 3]
```diff
@@ -70,7 +204,7 @@ public async Task Send_request_from_Ice_to_IceRpc([Values] bool oneway)
                 cancellationToken);
             tcs.SetResult(readResult.Buffer.ToArray());
             request.Payload.AdvanceTo(readResult.Buffer.End);
-            return new OutgoingResponse(request);
+            return new IceRpc.OutgoingResponse(request);
```
Comment (bernardnormier) [inline tests/Interop.Tests/ConnectionTests.cs:207]: Ice 3.8 has its own OutgoingResponse.

File: tests/Interop.Tests/Slice/TagTests.ice [hunk 4]
```diff
@@ -56,8 +51,6 @@ module Interop
                     optional(1) OneByteSeq oneByteSeq,   // tag type = VSize (+ optimization)
                     optional(5) PointSeq pointSeq,       // tag type = VSize
                     optional(10) StringSeq stringSeq);   // tag type = FSize
-
-                void opClass(optional(1) Shape shape); // tag type = Class
```
Comment (bernardnormier) [inline tests/Interop.Tests/Slice/TagTests.ice:60]: Ice 3.8 does not support optional class parameters.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

**Pull request merged**